integration-tests/mock-components/conflicting-global-counter.js

```diff
@@ -0,0 +1,23 @@
+const { registerHtml, useGlobalObservable } = require('../../src/tram-one')
+const html = registerHtml()
+
+module.exports = () => {
+	// example where it is undefined first
+	useGlobalObservable('global-counter', undefined)
+	const [counter] = useGlobalObservable('global-counter', 5)
+
+	// example where it is defined first
+	useGlobalObservable('global-counter-2', 5)
+	const [counter2] = useGlobalObservable('global-counter-2', undefined)
```

Comment:
these lines verify that regardless of the order, a global-observable will not be overridden with `undefined`. This is more common when we want to depend on one component setting a value, and somewhere else in the code we want to just read it (and we have no idea what the default value might be).

```javascript
// somewhere.js
const [setupValue] = useGlobalObservable('global-value', 5)
```

```javascript
// somewhere-else.js
const [readValue] = useGlobalObservable('global-value')
```